x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_list_page/index.tsx

```diff
@@ -62,6 +64,11 @@ export const AgentListPage: React.FunctionComponent<{}> = () => {
   useBreadcrumbs('agent_list');
   const defaultKuery: string = (useUrlParams().urlParams.kuery as string) || '';
 
+  // Missing encryption key callout
+  const { missingOptionalFeatures } = useFleetStatus();
```

Comment:
Could we move this inside the `useMissingEncryptionKeyCalloutHasBeenDismissed` hook? So if we want to reuse that hook, we don't have to check for this value since it's already used in the hook.

When moving this, you will have to also move the `showMissingEncryptKeyCallout`, so the hook can return if show or not the callout depending on the encryptionKey and the dismiss callout value.

Thoughts?

Reply:
If you do the above change, you can then rename the hook to: `useMissingEncryptionKeyCallout`, that returns a `canShowMissingEncryptionKeyCallout` boolean and `dismissEncryptionKeyCallout()` function